### ISSUE TITLE
`--build-arg` and `ARG` not working when the `ARG` used before `FROM`

### ISSUE THREAD

**Issue by TriplEight** [containers/buildah#3531]
Apparently, this didn't work in this case:

```
FROM docker.io/library/node:14

# ...
ARG REGISTRY_PATH=docker.io/myorg

FROM ${REGISTRY_PATH}/image:latest
# ...
```

Neither `ARG REGISTRY_PATH=docker.io/myorg` nor --build-arg="docker.io/myorg" didn't affect the second `FROM` and it was failing with:

```
...
STEP 5: FROM /base-ci:latest
error creating build container: Invalid image name "/image:latest", unknown transport "/image"
time="2021-09-21T18:30:55Z" level=error msg="exit status 125"
Error: Error: buildah exited with code 125
```

_Originally posted by @TriplEight in https://github.com/containers/buildah/issues/1500#issuecomment-924815564_

**Comment by nalind** [comment]
I didn't think that was expected to work.  The build arg created with `ARG REGISTRY_PATH=docker.io/myorg` value would not have an effect on the `FROM` instruction because it's not in the same stage as the `FROM ${REGISTRY_PATH}/image:latest` instruction, which is the start of the next stage.

**Comment by TriplEight** [comment]
@nalind good point, might be. But in this case `--build-arg` doesn't work either, is it because it's being intercepted by the `ARG` with the same var? Then I think such a corner case should be addressed somehow.

**Comment by nalind** [comment]
`ARG`s are generally only usable in a stage after they're defined for that stage, so at the point where the `FROM` instruction that starts a stage is being parsed, the argument isn't known to be a valid argument.  It would have to be defined again in the stage in order to be usable by later instructions in that stage.
Heading `ARG`s (those defined before the first `FROM` instruction) should be usable in the `FROM` instructions for any of the build's stages.
